### PR TITLE
Add features to account to allow wasm compilation

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
# Description of change

It should be possible to compile the upcoming identity actor to WebAssembly. It will use types from the account crate, and the account crate is not Wasm compatible due to a dependency on stronghold. This PR adds features to the account crate to opt-in to stronghold support - which is enabled by default - but can be opted out of to enable compilation to Wasm.

There is a weird issue with `tokio`, where even if we specify it to only use feature `sync` for example, it will still depend on the `mio` crate, even though it shouldn't. This may be happening due to how cargo selects features, and I've described the issue in some detail in the `toml` file. Fortunately, this can be worked around by specifying dependencies via git rather than crates.io. Makes it a bit less ergonomic to work with, i.e. we have to explicitly specify when we want to update `tokio` by updating the revision, but that shouldn't be too much of a hassle.

## Links to any relevant issues
Be sure to reference any related issues by adding `fixes issue #`.

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
Describe the tests that you ran to verify your changes.
Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-account/Cargo.toml [hunk 1]
```diff
@@ -17,16 +17,19 @@ hashbrown = { version = "0.9", features = ["serde"] }
...
+identity-iota = { version = "=0.3.0", path = "../identity-iota", default-features = false }
 itoa = { version = "0.4" }
 log = { version = "0.4", default-features = false }
 once_cell = { version = "1.7", default-features = false, features = ["std"] }
 paste = { version = "1.0" }
-riker = { version = "0.4" }
-serde = { version = "1.0", default-features = false, features = ["alloc", "derive"] }
+riker = { version = "0.4", optional = true }
+serde = { version = "1.0", default-features = false, features = [
+    "alloc",
```
Comment (l1h3r) [inline identity-account/Cargo.toml:27]: 💅 strange formatting 😛

File: identity-account/Cargo.toml [hunk 2]
```diff
@@ -17,16 +17,19 @@ hashbrown = { version = "0.9", features = ["serde"] }
...
+    "derive"
+] }
 slog = { version = "2.7" }
 thiserror = { version = "1.0" }
-tokio = { version = "1.5", default-features = false, features = ["rt", "rt-multi-thread", "sync"] }
+tokio = { version = "1.7", features = ["sync"] }
```
Comment (l1h3r) [inline identity-account/Cargo.toml:32]: was the change to 1.7 required to fix wasm compilation? maybe better to update as a separate PR.

File: identity-account/Cargo.toml [hunk 3]
```diff
@@ -36,15 +39,30 @@ features = ["blake2b", "ed25519", "hmac", "pbkdf", "sha", "slip10", "std"]
...
 [features]
 mem-client = []
+stronghold = [
+    "iota_stronghold",
+    "stronghold_engine",
+    "riker",
+    "tokio/rt-multi-thread",
+]
+wasm = ["identity-iota/wasm"]
+default = ["stronghold", "identity-iota/async"]
```
Comment (l1h3r) [inline identity-account/Cargo.toml:68]: better to have an `async` feature that enables this feature in `identity-iota` (`default = ["stronghold", "async"]`). at the moment it's a bit awkward to disable stronghold and enable the `async` feature

File: identity-account/Cargo.toml (hunk 2 above)
Comment (l1h3r) [reply identity-account/Cargo.toml:32]: in any case, the tokio version in the examples should be kept in sync with this one.

Comment (PhilippGackstatter) [reply identity-account/Cargo.toml:32]: No that wasn't required for Wasm, I just upgraded it. But should be a separate PR, I agree.

**Review by JelleMillenaar — APPROVED** [review]

**Pull request merged**